liesel/goose/gibbs.py

```diff
@@ -123,3 +136,85 @@ def end_warmup(
         """
 
         return WarmupOutcome(error_code=0, kernel_state=kernel_state)
+
+
+def create_categorical_gibbs_kernel(
+    name: str, outcomes: Sequence[Array], model: "Model"
+) -> GibbsKernel:
+    """
+    Creates a categorical Gibbs kernel.
+
+    The prior distribution of the variable to sample must be a categorical distribution,
+    usually implemented via :class:`tfd.FiniteDiscrete`.
+
+    This kernel evaluates the full conditional log probability of the model for each
+    possible value of the variable to sample. It then draws a new value for the variable
+    from the categorical distribution defined by the full conditional log probabilities.
+
+    Usually, you can define more efficient specialized kernels if you know the actual
+    full conditional distribution of the variable to sample.
+
+    Parameters
+    ----------
+    name
+        The name of the variable to sample.
+    outcomes
+        List of possible outcomes.
+    model
+        The model to sample from.
+
+    Examples
+    --------
+
+    In the following example, we create a categorical Gibbs kernel for a variable
+    with three possible values. The prior distribution of the variable is a categorical
+    distribution with probabilities ``[0.1, 0.2, 0.7]``.
+
+    You can then use the kernel to sample from the model.
+
+    >>> import tensorflow_probability.substrates.jax.distributions as tfd
+
+    >>> values = [0, 1, 2]
+    >>> prior_probs = [0.1, 0.2, 0.7]
+    >>> value_grid = lsl.Var(values, name="value_grid")
+
+    >>> prior = lsl.Dist(tfd.FiniteDiscrete, outcomes=value_grid, probs=prior_probs)
+    >>> categorical_var = lsl.Var(
+    ...     value=values[0],
+    ...     distribution=prior,
+    ...     name="categorical_var",
+    ... )
+
+    >>> model = lsl.GraphBuilder().add(categorical_var).build_model()
+    >>> kernel = create_categorical_gibbs_kernel("categorical_var", values, model)
+    >>> type(kernel)
+    <class 'liesel.goose.gibbs.GibbsKernel'>
+
+    """
+
+    model = model._copy_computational_model()
+
+    def transition_fn(prng_key, model_state):
+
+        model.state = model_state
+        for node in model.nodes.values():
+            node._outdated = False
+
+        def conditional_log_prob_fn(value: Array):
+            """
+            Evaluates the full conditional log probability of the model
+            given the input value.
+            """
+            model.vars[name].value = value
```

Comment:
How about using `model.auto_update = False` and `model.update("_model_log_prob")` here for better efficiency? This way, only the model log-prob node and its recursive inputs would be updated.